### PR TITLE
Address comments from #2302

### PR DESCRIPTION
**Pull request by domenukk**
A bunch of smaller cleanups addressing my comments in PR #2302

### CODE REVIEW

File: libafl/src/events/launcher.rs [hunk 1]
```diff
@@ -744,11 +742,16 @@ where
 
         #[cfg(feature = "multi_machine")]
         // Create this after forks, to avoid problems with tokio runtime
-        let (multi_machine_sender_hook, multi_machine_receiver_hook) =
+
+        // # Safety
+        // The `multi_machine_receiver_hook` needs messages to outlive the receiver.
+        // The underlying memory region for incoming messages lives longer than the async thread processing them.
```
Comment (domenukk) [inline libafl/src/events/launcher.rs:748]: @rmalmain Are we sure this is the case? Is there any race condition when memory gets cleared up before the async thread kicks in?

Comment (rmalmain) [reply libafl/src/events/launcher.rs:748]: afaik the messages live long enough. i'll double check this

**Comment by domenukk** [comment]
@rmalmain can you take a look if you're happy with these changes?

**Comment by rmalmain** [comment]
@domenukk looks good to me, thanks for cleaning up

File: libafl/src/events/tcp.rs [hunk 2]
```diff
@@ -0,0 +1,1364 @@
...
+    SP: ShMemProvider,
+    S: State + HasExecutions + HasMetadata,
+    MT: Monitor + Clone,
+{
+    /// Launch the restarting manager
+    pub fn launch(&mut self) -> Result<(Option<S>, TcpRestartingEventManager<EMH, S, SP>), Error> {
+        // We start ourself as child process to actually fuzz
+        let (staterestorer, _new_shmem_provider, core_id) = if env::var(_ENV_FUZZER_SENDER).is_err()
+        {
+            let broker_things = |mut broker: TcpEventBroker<S::Input, MT>, _remote_broker_addr| {
```
Comment (tokatoka) [inline libafl/src/events/tcp.rs:1149]: this is not really working for remote host...
this is the only place that remote_broker_addr is every passed, and it completely ignores the address

Comment (domenukk) [reply libafl/src/events/tcp.rs:1149]: Yes would be good to fix it at some point

**Pull request merged**